File: spec/cpp_stl/test_nav_parent3.cpp
#include <boost/test/unit_test.hpp>

#include <nav_parent3.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_nav_parent3) {
    std::ifstream ifs("src/nav_parent2.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    nav_parent3_t* r = new nav_parent3_t(&ks);

    BOOST_CHECK_EQUAL(r->ofs_tags(), 8);
    BOOST_CHECK_EQUAL(r->num_tags(), 2);

    BOOST_CHECK_EQUAL(r->tags()->at(0)->name(), "RAHC");
    BOOST_CHECK_EQUAL(r->tags()->at(0)->ofs(), 0x20);
    BOOST_CHECK_EQUAL(r->tags()->at(0)->num_items(), 3);
    BOOST_CHECK_EQUAL(r->tags()->at(0)->tag_content()->content(), "foo");

    BOOST_CHECK_EQUAL(r->tags()->at(1)->name(), "RAHC");
    BOOST_CHECK_EQUAL(r->tags()->at(1)->ofs(), 0x23);
    BOOST_CHECK_EQUAL(r->tags()->at(1)->num_items(), 6);
    BOOST_CHECK_EQUAL(r->tags()->at(1)->tag_content()->content(), "barbaz");

    delete r;
}

File: spec/java/src/io/kaitai/struct/spec/TestNavParent3.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.NavParent3;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestNavParent3 extends CommonSpec {
    @Test
    public void testNavParent3() throws Exception {
        NavParent3 r = NavParent3.fromFile(SRC_DIR + "nav_parent2.bin");

        assertEquals(r.ofsTags(), 8);
        assertEquals(r.numTags(), 2);

        assertEquals(r.tags().get(0).name(), "RAHC");
        assertEquals(r.tags().get(0).ofs(), 0x20);
        assertEquals(r.tags().get(0).numItems(), 3);
        assertEquals(r.tags().get(0).tagContent().content(), "foo");

        assertEquals(r.tags().get(1).name(), "RAHC");
        assertEquals(r.tags().get(1).ofs(), 0x23);
        assertEquals(r.tags().get(1).numItems(), 6);
        assertEquals(r.tags().get(1).tagContent().content(), "barbaz");
    }
}
